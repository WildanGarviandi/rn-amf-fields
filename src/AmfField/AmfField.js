import React, { Component } from 'react'
import AmfTextInput from '../AmfTextInput/AmfTextInput'
import AmfLikertScale from '../AmfLikertScale/AmfLikertScale'
import AmfNumberInput from '../AmfNumberInput/AmfNumberInput'
import AmfFormattedNumberInput from '../AmfFormattedNumberInput/AmfFormattedNumberInput'
import AmfRadioButton from '../AmfRadioButton/AmfRadioButton'
import AmfRadioButtonVertical from '../AmfRadioButton/AmfRadioButtonVertical'
import AmfDatepicker from '../AmfDatepicker/AmfDatepicker'
import AmfDateInput from '../AmfDateInput/AmfDateInput'
import AmfSelect from '../AmfSelect/AmfSelect'
import AmfSelectNested from '../AmfSelectNested/AmfSelectNested'
import PropTypes from 'prop-types'
import fieldPropTypes from '../fieldPropTypes'

const switcher = {
  'text': AmfTextInput,
  'likert': AmfLikertScale,
  'number': AmfNumberInput,
  'formatted-number': AmfFormattedNumberInput,
  'select': AmfSelect,
  'select-nested': AmfSelectNested,
  'radio' : AmfRadioButton,
  'radio-vertical' : AmfRadioButtonVertical,
  'datepicker': AmfDatepicker,
  'date': AmfDateInput,
}

class AmfField extends Component {

  constructor(props) {
    super(props)
  }

  render() {
    const { type, label, onChange, value, options  } = this.props
    if (type in switcher) {
      const Component = switcher[type]
      return <Component type={type} label={label} onChange={onChange} value={value} options={options} />
    } else {
      throw new TypeError ('Unknown type Field "' + type + '" in AmfField')
    }
  }
}


AmfField.propTypes = {...fieldPropTypes, type:PropTypes.string.isRequired}

export default AmfField
