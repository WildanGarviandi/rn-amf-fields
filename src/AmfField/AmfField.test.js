import React from 'react'
import AmfField from './AmfField'
import renderer from 'react-test-renderer'

test('AmfField without validation', () => {
  let props = { 
    type  : 'text',
    label : 'Label satu dua tiga',
    onChange : () => null,
    value : 'Satu'
  }

  const amfField = <AmfField {...props} />
  expect(amfField).not.toBe(undefined)

  const component = renderer.create(amfField)
  
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})


test('AmfField should be able to render AmfSelect (should transfer its options)', () => {
  let props = { 
    type  : 'select',
    label : 'Label satu dua tiga',
    onChange : () => null,
    value : 'Satu',
    options : [
      {label:'Label 1', value:'Satu'},
      {label:'Label 2', value:'Dua'},
      {label:'Label 3', value:'Tiga'},
      {label:'Label 4', value:'Empat'}
    ]
  }

  const amfField = <AmfField {...props} />
  expect(amfField).not.toBe(undefined)

  const component = renderer.create(amfField)
  
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('AmfField should be able to render AmfLikertScale (should transfer its options)', () => {
  let props = { 
    type  : 'likert',
    label : 'Label satu dua tiga',
    onChange : () => null,
    value : 'Satu',
    options : [
      {label:'Label 1', value:'Satu'},
      {label:'Label 2', value:'Dua'},
      {label:'Label 3', value:'Tiga'},
      {label:'Label 4', value:'Empat'}
    ]
  }

  const amfField = <AmfField {...props} />
  expect(amfField).not.toBe(undefined)

  const component = renderer.create(amfField)
  
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

/**
 * it should be last because console.error mock error
 *
 */
test('AmfField with wrong type', ()  => {
  let props = { 
    type  : 'zzzzzzzzzzzzzzzzzzzzzzzz',
    label : 'Label satu dua tiga',
    onChange : () => null,
    value : 'Satu'
  }

  console.error = jest.fn() 
  //const errorMessage = 'Unknown type Field "' + props.type + '" in AmfField';
  expect( () => renderer.create(<AmfField {...props} />)).toThrowErrorMatchingSnapshot()
  expect(console.error).toBeCalled()
})
