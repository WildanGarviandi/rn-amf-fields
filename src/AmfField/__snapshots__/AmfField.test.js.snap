// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AmfField should be able to render AmfLikertScale (should transfer its options) 1`] = `
<View
  style={
    Object {
      "marginVertical": 8,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#454545",
        "marginBottom": 8,
        "marginRight": 16,
      }
    }
  >
    Label satu dua tiga
  </Text>
  <View
    style={
      Array [
        Object {
          "padding": 8,
        },
        Object {},
      ]
    }
  >
    <View
      onLayout={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-around",
          "marginBottom": 16,
        }
      }
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "rgb(165, 0, 38)",
            "borderColor": "rgb(165, 0, 38)",
            "borderRadius": 0,
            "borderWidth": 1,
            "height": 0,
            "justifyContent": "center",
            "opacity": 1,
            "width": 0,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "textAlign": "center",
              },
              Object {
                "color": "white",
                "fontSize": 0,
              },
            ]
          }
        >
          Satu
        </Text>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "white",
            "borderColor": "rgb(248, 141, 82)",
            "borderRadius": 0,
            "borderWidth": 1,
            "height": 0,
            "justifyContent": "center",
            "opacity": 1,
            "width": 0,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "textAlign": "center",
              },
              Object {
                "color": "rgb(248, 141, 82)",
                "fontSize": 0,
              },
            ]
          }
        >
          Dua
        </Text>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "white",
            "borderColor": "rgb(133, 203, 103)",
            "borderRadius": 0,
            "borderWidth": 1,
            "height": 0,
            "justifyContent": "center",
            "opacity": 1,
            "width": 0,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "textAlign": "center",
              },
              Object {
                "color": "rgb(133, 203, 103)",
                "fontSize": 0,
              },
            ]
          }
        >
          Tiga
        </Text>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "white",
            "borderColor": "rgb(0, 104, 55)",
            "borderRadius": 0,
            "borderWidth": 1,
            "height": 0,
            "justifyContent": "center",
            "opacity": 1,
            "width": 0,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "textAlign": "center",
              },
              Object {
                "color": "rgb(0, 104, 55)",
                "fontSize": 0,
              },
            ]
          }
        >
          Empat
        </Text>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
          },
          Object {
            "marginHorizontal": 0,
          },
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "textAlign": "center",
            "width": 0,
          }
        }
      >
        Label 1
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "textAlign": "center",
            "width": 0,
          }
        }
      >
        Label 4
      </Text>
    </View>
  </View>
</View>
`;

exports[`AmfField should be able to render AmfSelect (should transfer its options) 1`] = `
<View>
  <View
    onLayout={[Function]}
    style={undefined}
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      pointerEvents="box-only"
      style={undefined}
      testID={undefined}
    >
      <View
        onResponderRelease={[Function]}
        onStartShouldSetResponder={[Function]}
        pointerEvents="none"
        style={undefined}
      >
        <View
          collapsable={undefined}
          style={
            Object {
              "backgroundColor": "transparent",
              "borderBottomColor": "rgba(0, 0, 0, 0.38)",
              "borderBottomWidth": 0.5,
              "height": 64,
              "paddingBottom": 8,
              "paddingTop": 32,
            }
          }
        >
          <View
            collapsable={undefined}
            style={
              Object {
                "position": "absolute",
                "top": 16,
              }
            }
          >
            <Text
              accessible={true}
              active={true}
              allowFontScaling={true}
              animationDuration={225}
              collapsable={undefined}
              disabled={false}
              ellipsizeMode="tail"
              errored={false}
              focused={false}
              numberOfLines={1}
              style={
                Object {
                  "color": "rgba(0, 0, 0, 0.38)",
                  "fontSize": 12,
                }
              }
            >
              Label satu dua tiga
            </Text>
          </View>
          <View
            style={
              Object {
                "flexDirection": "row",
              }
            }
          >
            <TextInput
              autoCapitalize="sentences"
              blurOnSubmit={true}
              data={
                Array [
                  Object {
                    "label": "Label 1",
                    "value": "Satu",
                  },
                  Object {
                    "label": "Label 2",
                    "value": "Dua",
                  },
                  Object {
                    "label": "Label 3",
                    "value": "Tiga",
                  },
                  Object {
                    "label": "Label 4",
                    "value": "Empat",
                  },
                ]
              }
              disableFullscreenUI={true}
              editable={false}
              error=""
              itemColor="rgba(0, 0, 0, .54)"
              itemCount={4}
              keyboardType={undefined}
              onBlur={[Function]}
              onChange={[Function]}
              onChangeText={[Function]}
              onContentSizeChange={[Function]}
              onFocus={[Function]}
              renderAccessory={[Function]}
              returnKeyType={undefined}
              selectionColor="rgb(0, 145, 234)"
              shadeOpacity={0.12}
              style={
                Array [
                  Object {
                    "flex": 1,
                    "margin": 0,
                    "padding": 0,
                    "top": 2,
                  },
                  Object {
                    "color": "rgba(0, 0, 0, .87)",
                    "fontSize": 16,
                    "height": 24,
                  },
                  undefined,
                ]
              }
              underlineColorAndroid="transparent"
              value="Satu"
            />
            <View
              style={
                Object {
                  "alignSelf": "flex-start",
                  "justifyContent": "center",
                  "top": 2,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "height": 24,
                    "justifyContent": "center",
                    "width": 24,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "transparent",
                      "height": 6,
                      "overflow": "hidden",
                      "width": 12,
                    }
                  }
                >
                  <View
                    style={
                      Array [
                        Object {
                          "height": 8,
                          "transform": Array [
                            Object {
                              "translateY": -4,
                            },
                            Object {
                              "rotate": "45deg",
                            },
                          ],
                          "width": 8,
                        },
                        Object {
                          "backgroundColor": "rgba(0, 0, 0, .38)",
                        },
                      ]
                    }
                  />
                </View>
              </View>
            </View>
          </View>
        </View>
        <View
          collapsable={undefined}
          style={
            Object {
              "flexDirection": "row",
              "height": 8,
            }
          }
        >
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "bottom": 0,
                  "left": 0,
                  "paddingVertical": 4,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                collapsable={undefined}
                disabled={false}
                ellipsizeMode="tail"
                numberOfLines={1}
                style={
                  Object {
                    "backgroundColor": "transparent",
                    "color": "rgb(213, 0, 0)",
                    "fontSize": 0,
                    "opacity": 0,
                  }
                }
              >
                
              </Text>
            </View>
            <View
              style={
                Object {
                  "bottom": 0,
                  "left": 0,
                  "paddingVertical": 4,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                collapsable={undefined}
                disabled={false}
                ellipsizeMode="tail"
                numberOfLines={1}
                style={
                  Object {
                    "backgroundColor": "transparent",
                    "color": "rgba(0, 0, 0, .38)",
                    "fontSize": 12,
                    "opacity": 1,
                  }
                }
              />
            </View>
          </View>
        </View>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hitSlop={undefined}
        nativeID={undefined}
        onLayout={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        pointerEvents="box-only"
        rippleCentered={false}
        rippleColor="rgba(0, 0, 0, .38)"
        rippleDuration={450}
        rippleOpacity={0.54}
        rippleSequential={true}
        rippleSize={0}
        style={
          Object {
            "height": 48,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 16,
          }
        }
        testID={undefined}
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "bottom": 0,
                "left": 0,
                "overflow": "hidden",
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              Object {
                "borderRadius": 0,
              },
            ]
          }
        />
      </View>
    </View>
    <Modal
      hardwareAccelerated={false}
      onRequestClose={[Function]}
      transparent={true}
      visible={false}
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "height": 1334,
            "width": 750,
          }
        }
        testID={undefined}
      >
        <View
          collapsable={undefined}
          style={
            Object {
              "backgroundColor": "rgb(255, 255, 255)",
              "borderRadius": 2,
              "height": 176,
              "left": undefined,
              "opacity": 0,
              "position": "absolute",
              "shadowColor": "rgba(0, 0, 0, 1.0)",
              "shadowOffset": Object {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 0.54,
              "shadowRadius": 2,
              "top": undefined,
              "transform": Array [
                Object {
                  "translateY": -48,
                },
              ],
              "width": undefined,
            }
          }
        >
          <RCTScrollView
            contentContainerStyle={
              Object {
                "paddingVertical": 8,
              }
            }
            scrollEnabled={false}
            style={
              Object {
                "borderRadius": 2,
                "flex": 1,
              }
            }
          >
            <View>
              <View
                accessibilityComponentType={undefined}
                accessibilityLabel={undefined}
                accessibilityTraits={undefined}
                accessible={true}
                collapsable={undefined}
                focusAnimation={null}
                focusAnimationDuration={225}
                fontSize={16}
                hitSlop={
                  Object {
                    "bottom": 6,
                    "left": 4,
                    "right": 4,
                    "top": 6,
                  }
                }
                index={0}
                nativeID={undefined}
                onLayout={[Function]}
                onPress={[Function]}
                onPressIn={[Function]}
                onPressOut={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                pointerEvents="box-only"
                rippleCentered={false}
                rippleColor="rgba(0, 0, 0, .38)"
                rippleDuration={450}
                rippleOpacity={0.54}
                rippleSequential={true}
                rippleSize={0}
                style={
                  Object {
                    "backgroundColor": "transparent",
                    "borderRadius": 0,
                    "height": 40,
                    "justifyContent": "center",
                    "paddingLeft": undefined,
                    "paddingRight": undefined,
                    "paddingVertical": 8,
                  }
                }
                testID={undefined}
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  disabled={false}
                  ellipsizeMode="tail"
                  numberOfLines={1}
                  style={
                    Array [
                      undefined,
                      Object {
                        "color": "rgba(0, 0, 0, .87)",
                        "fontSize": 16,
                      },
                    ]
                  }
                >
                  Satu
                </Text>
                <View
                  collapsable={undefined}
                  style={
                    Object {
                      "backgroundColor": "rgba(0, 0, 0, .38)",
                      "borderRadius": 0,
                      "bottom": 0,
                      "left": 0,
                      "opacity": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    }
                  }
                />
                <View
                  style={
                    Array [
                      Object {
                        "backgroundColor": "transparent",
                        "bottom": 0,
                        "left": 0,
                        "overflow": "hidden",
                        "position": "absolute",
                        "right": 0,
                        "top": 0,
                      },
                      Object {
                        "borderRadius": 0,
                      },
                    ]
                  }
                />
              </View>
              <View
                accessibilityComponentType={undefined}
                accessibilityLabel={undefined}
                accessibilityTraits={undefined}
                accessible={true}
                collapsable={undefined}
                focusAnimation={null}
                focusAnimationDuration={225}
                fontSize={16}
                hitSlop={
                  Object {
                    "bottom": 6,
                    "left": 4,
                    "right": 4,
                    "top": 6,
                  }
                }
                index={1}
                nativeID={undefined}
                onLayout={[Function]}
                onPress={[Function]}
                onPressIn={[Function]}
                onPressOut={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                pointerEvents="box-only"
                rippleCentered={false}
                rippleColor="rgba(0, 0, 0, .38)"
                rippleDuration={450}
                rippleOpacity={0.54}
                rippleSequential={true}
                rippleSize={0}
                style={
                  Object {
                    "backgroundColor": "transparent",
                    "borderRadius": 0,
                    "height": 40,
                    "justifyContent": "center",
                    "paddingLeft": undefined,
                    "paddingRight": undefined,
                    "paddingVertical": 8,
                  }
                }
                testID={undefined}
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  disabled={false}
                  ellipsizeMode="tail"
                  numberOfLines={1}
                  style={
                    Array [
                      undefined,
                      Object {
                        "color": "rgba(0, 0, 0, .87)",
                        "fontSize": 16,
                      },
                    ]
                  }
                >
                  Dua
                </Text>
                <View
                  collapsable={undefined}
                  style={
                    Object {
                      "backgroundColor": "rgba(0, 0, 0, .38)",
                      "borderRadius": 0,
                      "bottom": 0,
                      "left": 0,
                      "opacity": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    }
                  }
                />
                <View
                  style={
                    Array [
                      Object {
                        "backgroundColor": "transparent",
                        "bottom": 0,
                        "left": 0,
                        "overflow": "hidden",
                        "position": "absolute",
                        "right": 0,
                        "top": 0,
                      },
                      Object {
                        "borderRadius": 0,
                      },
                    ]
                  }
                />
              </View>
              <View
                accessibilityComponentType={undefined}
                accessibilityLabel={undefined}
                accessibilityTraits={undefined}
                accessible={true}
                collapsable={undefined}
                focusAnimation={null}
                focusAnimationDuration={225}
                fontSize={16}
                hitSlop={
                  Object {
                    "bottom": 6,
                    "left": 4,
                    "right": 4,
                    "top": 6,
                  }
                }
                index={2}
                nativeID={undefined}
                onLayout={[Function]}
                onPress={[Function]}
                onPressIn={[Function]}
                onPressOut={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                pointerEvents="box-only"
                rippleCentered={false}
                rippleColor="rgba(0, 0, 0, .38)"
                rippleDuration={450}
                rippleOpacity={0.54}
                rippleSequential={true}
                rippleSize={0}
                style={
                  Object {
                    "backgroundColor": "transparent",
                    "borderRadius": 0,
                    "height": 40,
                    "justifyContent": "center",
                    "paddingLeft": undefined,
                    "paddingRight": undefined,
                    "paddingVertical": 8,
                  }
                }
                testID={undefined}
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  disabled={false}
                  ellipsizeMode="tail"
                  numberOfLines={1}
                  style={
                    Array [
                      undefined,
                      Object {
                        "color": "rgba(0, 0, 0, .87)",
                        "fontSize": 16,
                      },
                    ]
                  }
                >
                  Tiga
                </Text>
                <View
                  collapsable={undefined}
                  style={
                    Object {
                      "backgroundColor": "rgba(0, 0, 0, .38)",
                      "borderRadius": 0,
                      "bottom": 0,
                      "left": 0,
                      "opacity": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    }
                  }
                />
                <View
                  style={
                    Array [
                      Object {
                        "backgroundColor": "transparent",
                        "bottom": 0,
                        "left": 0,
                        "overflow": "hidden",
                        "position": "absolute",
                        "right": 0,
                        "top": 0,
                      },
                      Object {
                        "borderRadius": 0,
                      },
                    ]
                  }
                />
              </View>
              <View
                accessibilityComponentType={undefined}
                accessibilityLabel={undefined}
                accessibilityTraits={undefined}
                accessible={true}
                collapsable={undefined}
                focusAnimation={null}
                focusAnimationDuration={225}
                fontSize={16}
                hitSlop={
                  Object {
                    "bottom": 6,
                    "left": 4,
                    "right": 4,
                    "top": 6,
                  }
                }
                index={3}
                nativeID={undefined}
                onLayout={[Function]}
                onPress={[Function]}
                onPressIn={[Function]}
                onPressOut={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                pointerEvents="box-only"
                rippleCentered={false}
                rippleColor="rgba(0, 0, 0, .38)"
                rippleDuration={450}
                rippleOpacity={0.54}
                rippleSequential={true}
                rippleSize={0}
                style={
                  Object {
                    "backgroundColor": "transparent",
                    "borderRadius": 0,
                    "height": 40,
                    "justifyContent": "center",
                    "paddingLeft": undefined,
                    "paddingRight": undefined,
                    "paddingVertical": 8,
                  }
                }
                testID={undefined}
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  disabled={false}
                  ellipsizeMode="tail"
                  numberOfLines={1}
                  style={
                    Array [
                      undefined,
                      Object {
                        "color": "rgba(0, 0, 0, .87)",
                        "fontSize": 16,
                      },
                    ]
                  }
                >
                  Empat
                </Text>
                <View
                  collapsable={undefined}
                  style={
                    Object {
                      "backgroundColor": "rgba(0, 0, 0, .38)",
                      "borderRadius": 0,
                      "bottom": 0,
                      "left": 0,
                      "opacity": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    }
                  }
                />
                <View
                  style={
                    Array [
                      Object {
                        "backgroundColor": "transparent",
                        "bottom": 0,
                        "left": 0,
                        "overflow": "hidden",
                        "position": "absolute",
                        "right": 0,
                        "top": 0,
                      },
                      Object {
                        "borderRadius": 0,
                      },
                    ]
                  }
                />
              </View>
            </View>
          </RCTScrollView>
        </View>
      </View>
    </Modal>
  </View>
</View>
`;

exports[`AmfField with wrong type 1`] = `"Unknown type Field \\"zzzzzzzzzzzzzzzzzzzzzzzz\\" in AmfField"`;

exports[`AmfField without validation 1`] = `
<View>
  <View
    onResponderRelease={[Function]}
    onStartShouldSetResponder={[Function]}
    pointerEvents="auto"
    style={undefined}
  >
    <View
      collapsable={undefined}
      style={
        Object {
          "backgroundColor": "transparent",
          "borderBottomColor": "rgba(0, 0, 0, 0.38)",
          "borderBottomWidth": 0.5,
          "height": 64,
          "paddingBottom": 8,
          "paddingTop": 32,
        }
      }
    >
      <View
        collapsable={undefined}
        style={
          Object {
            "position": "absolute",
            "top": 16,
          }
        }
      >
        <Text
          accessible={true}
          active={true}
          allowFontScaling={true}
          animationDuration={225}
          collapsable={undefined}
          disabled={false}
          ellipsizeMode="tail"
          errored={false}
          focused={false}
          numberOfLines={1}
          style={
            Object {
              "color": "rgba(0, 0, 0, 0.38)",
              "fontSize": 12,
            }
          }
        >
          Label satu dua tiga
        </Text>
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <TextInput
          autoCapitalize="sentences"
          blurOnSubmit={true}
          disableFullscreenUI={true}
          editable={true}
          keyboardType="default"
          onBlur={[Function]}
          onChange={[Function]}
          onChangeText={[Function]}
          onContentSizeChange={[Function]}
          onFocus={[Function]}
          options={Object {}}
          returnKeyType="next"
          selectionColor="rgb(0, 145, 234)"
          style={
            Array [
              Object {
                "flex": 1,
                "margin": 0,
                "padding": 0,
                "top": 2,
              },
              Object {
                "color": "rgba(0, 0, 0, .87)",
                "fontSize": 16,
                "height": 24,
              },
              undefined,
            ]
          }
          underlineColorAndroid="transparent"
          value="Satu"
        />
      </View>
    </View>
    <View
      collapsable={undefined}
      style={
        Object {
          "flexDirection": "row",
          "height": 8,
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <View
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "paddingVertical": 4,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            collapsable={undefined}
            disabled={false}
            ellipsizeMode="tail"
            numberOfLines={1}
            style={
              Object {
                "backgroundColor": "transparent",
                "color": "rgb(213, 0, 0)",
                "fontSize": 0,
                "opacity": 0,
              }
            }
          />
        </View>
        <View
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "paddingVertical": 4,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            collapsable={undefined}
            disabled={false}
            ellipsizeMode="tail"
            numberOfLines={1}
            style={
              Object {
                "backgroundColor": "transparent",
                "color": "rgba(0, 0, 0, .38)",
                "fontSize": 12,
                "opacity": 1,
              }
            }
          />
        </View>
      </View>
    </View>
  </View>
</View>
`;
